File: palindromo.c
#include <stdio.h>
#include <stdbool.h>
#include <time.h>
#include <stdlib.h>
#include <string.h>

bool palindromo(char *s)
{
    int length = strlen(s);

    for(int i = 0; i < length / 2; i++) // iteração por pares
    {
        if(s[i] != s[length - 1 - i])
        {
            return false; // não é palindromo
        } // end if 
    } // end for 
    return true; // é palindromo
} // end palindromo 

int main(void)
{
    // definir dados 
    bool result;
    char s[80];

    printf("Digite uma string: ");
    scanf("%s", s);
    
    // aplicar funções auxiliares
    result = palindromo(s);
    if(result == true)
    {
        printf("SIM\n");
    }
    else
    {
        printf("NAO\n");
    } // end if


    return 0;
} // end main 

File: palindromo.java
import java.util.Scanner;
import java.util.Locale;

public class Main
{
    boolean palindromo(String s)
    {
        for(int i = 0; i < s.length() / 2; i = i + 1) // iteração pelos pares 
        {
            int j = s.length() - 1 - i; // para cada indice i é calculado o indice correspondente na segunda metade
            if(s.charAt(i) != s.charAt(j))
            {
                return false; // não é um palindromo 
            } // end if 
        } // end for 
        return true; // é palindromo 
    } // end palindromo( )


    public static void main(String[] args)
    {
         Locale.setDefault(Locale.US);
         Scanner sc = new Scanner(System.in);
        
        // definir dados 
        String s; boolean result; 
        System.out.print("Digite uma string: ");
        s = sc.nextLine();

        // aplicar funções auxiliares
        Main mainObj = new Main(); // criar uma instância do objeto para chamar o método não estático
        result = mainObj.palindromo(s);

        if(result == true)
        {
            System.out.println("SIM");
        }
        else
        {
            System.out.println("NAO");
        } // end if 
         
         sc.close();
    } // end main( )
}